src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -622,40 +622,52 @@ static const uint8_t inverse_base64[256] = {
     255,
 };
 
-/* emission helpers */
-static grpc_error* on_hdr(grpc_chttp2_hpack_parser* p, grpc_mdelem md,
-                          int add_to_table) {
```

Comment:
I would make this a template param.  Basically, keep the `on_hdr_log()` method, but merge `on_hdr_common`, `on_hdr_no_add` and `on_addr_do_add`.

Reply:
Done.